### DISCUSSION TITLE
Add `_isCallerStreamSender` function

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#198] [Ideas]
As I mentioned in this [comment](https://github.com/sablierhq/v2-core/pull/196#issuecomment-1351377206), I suggest adding a function that returns whether the caller of a function is the sender or not. 

### Pros

- Would be possible to remove the `getSender` function from `SablierV2` contract and to change it to `external`
- Using the `isCallerSender` would be more gas efficient because it would return a boolean instead of an address, not by much tho 
- Consistency regarding checks in the `SablierV2` contract

**Comment by PaulRBerg** [comment]
This is an interesting idea, but I suggest we sleep on it for a few days before making a final decision.

My main hesitation is naming. Finding a good name for this ain't easy, because there is a bit of a terminology "congestion" here. As discussed in https://github.com/sablierhq/v2-core/discussions/81, it can sometimes be confusing what sender one is referring to - is it the Ethereum message caller (`msg.sender`) or the stream sender (it gets even more complicated when you consider that the sender of the stream doesn't necessarily have to be the funder of the stream!)?

Suggested alternative names for this function:

- `isCallerStreamSender`
- `isCallerSenderOfStream`

To respond to your stated pros.

- I agree that it would be nice to remove the `getSender` function from the abstract contract `SablierV2`, and mark it as `external`- this should save a little bit of gas, by the way. Furthermore, the caller sender function itself should be marked as `internal`, because no external user would find any value in calling this method independently.
- Returning a boolean - yes, that's cool, but, as you say, the gas save would be small.
- Regarding consistency. I'm afraid I am not quite sure what you mean by this. Could please re-articulate this point?

**Reply by andreivladbrg** [reply]
I prefer the `isCallerStreamSender` name

**Reply by PaulRBerg** [reply]
Good. Let's sleep on this idea some more before implementing it. There's a PR (https://github.com/sablierhq/v2-core/pull/201) that is waiting for your review now!

**Comment by PaulRBerg** [comment]
Locking, implemented in #205.
